common/src/main/java/com/sequenceiq/cloudbreak/eventbus/EventBus.java

```diff
@@ -0,0 +1,110 @@
+package com.sequenceiq.cloudbreak.eventbus;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executor;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import com.google.common.base.Preconditions;
+import com.google.common.eventbus.Subscribe;
+
+public class EventBus {
+
+    private final EventRouter eventRouter;
+
+    private final Executor executor;
+
+    private EventBus(EventRouter eventRouter, Executor executor) {
+        this.eventRouter = Preconditions.checkNotNull(eventRouter, "eventRouter must not be null.");
+        this.executor = Preconditions.checkNotNull(executor, "executor must not be null.");
+    }
+
+    public void notify(String key, Event<?> event) {
+        Preconditions.checkNotNull(key, "key must not be null.");
+        Preconditions.checkNotNull(event, "event must not be null.");
+        event.setKey(key);
+        executor.execute(() -> eventRouter.handle(event));
+    }
+
+    public <T extends Event<?>> void on(String key, Consumer<T> handler) {
+        Preconditions.checkNotNull(key, "key must not be null.");
+        Preconditions.checkNotNull(handler, "handler must not be null.");
+        eventRouter.addHandler(key, (Consumer<Event<?>>) handler);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    private static class EventRouter {
```

Comment:
why is this an inner class?

Reply:
It's an implementation detail of the EventBus that's why I didn't make it a public class.

Reply:
it would make the eventbus implementation cleaner and more readable if it was in a separate class. especially there is already a builder in the same.
I think both of them (bus and router) are missing unit tests